test/parallel/test-crypto-key-objects.js

```diff
@@ -200,6 +200,14 @@ const privateDsa = fixtures.readKey('dsa_private_encrypted_1025.pem',
     library: 'BIO routines',
     function: 'BIO_new_mem_buf',
   });
+
+  // This should not abort either: https://github.com/nodejs/node/issues/29904
+  assert.throws(() => {
+    createPrivateKey({ key: Buffer.alloc(0), format: 'der', type: 'spki' });
```

Comment:
There are other formats for public keys, perhaps its worth testing them all?

Reply:
I think the only other supported type is `pkcs1` which behaves differently, but I added a test case for that as well.